### PR TITLE
NEW! SuperText + SuperTextHeader + Pods Removal

### PR DESCRIPTION
**Pull request by rauchp**
- Fixed SigninScreen and ConfirmSignInScreen on Android
- Fixed drop shadow and super text in general for Android
- Made new SuperText with 3 types of adjustable drop shadows, and better adjustable shadow stroke, and, most importantly, reacts to View settings like space-between and space-around better.
- Upgraded a library (my new dev environment wouldn't work with RNReanimated  1.3.0)
- Deprecated RetroBrowseCardText, there is only one king now
- Standardized header heights 
- Fixed RoomView with long room names
- Removed Pods from repo.

New SuperText (OLD, see comment below for newer):
Screen Shot 2019-12-07 at 12 54 24 AM

Android Fix:
Screen Shot 2019-12-27 at 4 08 13 PM

Old Android SignIn:
Screen Shot 2019-12-07 at 12 54 24 AM

New Android SignIn (pre-fixing SuperText):
Screen Shot 2019-12-07 at 12 54 24 AM

Fixed Rooms w/ long names :
Simulator Screen Shot - iPhone 11

Headers are all consistent now:
Simulator Screen Shot - iPhone 11 - 2019-12-27 at 23 09 03

### CODE REVIEW

File: civilapp/ios/Podfile.lock [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PODS:
```
Comment (EphemeralX) [inline civilapp/ios/Podfile.lock]: Add most of this stuff to .gitignore.

```#Too big
ios/Pods/
ios/Podfile.lock
ios/Pods/Manifest.lock
ios/Pods/Pods.xcodeproj/project.pbxproj
ios/civilapp.xcodeproj/project.pbxproj
```

File: civilapp/src/auth/ProfileSelectScreen.js [hunk 2]
```diff
@@ -111,16 +107,26 @@ class ProfileSelectScreen extends Component {
   )
 
   renderHelpText = () => (
+
```
Comment (EphemeralX) [inline civilapp/src/auth/ProfileSelectScreen.js]: unneeded newline

File: civilapp/src/auth/ProfileSelectScreen.js [hunk 3]
```diff
@@ -111,16 +107,26 @@ class ProfileSelectScreen extends Component {
...
+      <SuperTextWithShadow
+        fontSize={14}
+        shadowSize={3}
+        containerStyle={{ marginBottom: 10 }}
+        fontFamily='Sportsball'
+        message='Choose your avatar'
+        mainColor='white'
+        mediumDropShadow
+      />
+
```
Comment (EphemeralX) [inline civilapp/src/auth/ProfileSelectScreen.js]: unneeded newline

File: civilapp/src/auth/ConfirmSignInScreen.js [hunk 4]
```diff
@@ -199,21 +199,8 @@ export default class ConfirmSignInScreen extends Component {
...
-        <View
-          style={{
-            flex: 1,
-            flexDirection: 'column',
-            alignItems: 'center',
-            justifyContent: 'flex-start',
-            paddingVertical: '10%'
-          }}>
+        style={styles.keyboardView}>
+
```
Comment (EphemeralX) [inline civilapp/src/auth/ConfirmSignInScreen.js]: unneeded newline

File: civilapp/src/auth/SignInScreen.js [hunk 5]
```diff
@@ -260,28 +246,31 @@ export default class SignInScreen extends Component {
     return (
       <RetroBackground>
         {this.renderHeader()}
+
```
Comment (EphemeralX) [inline civilapp/src/auth/SignInScreen.js]: unneeded newline

File: civilapp/src/auth/SignInScreen.js [hunk 6]
```diff
@@ -260,28 +246,31 @@ export default class SignInScreen extends Component {
     return (
       <RetroBackground>
         {this.renderHeader()}
+       
         {this.renderNumberInputView()}
         {this.renderNumberInputKeyboardView()}
+
```
Comment (EphemeralX) [inline civilapp/src/auth/SignInScreen.js]: unneeded newline

File: civilapp/src/components/SuperText.js [hunk 7]
```diff
@@ -17,310 +10,70 @@ const UPHEAVAL = Platform.OS === 'ios' ? 'Upheaval TT (BRK)' : 'upheavtt'
...
-        fontWeight: FONT_WEIGHT,
-        fontSize: props.fontSize,
-        color: props.mainColor ? props.mainColor : 'white',
-      }}
-    >
-      {props.message}
-    </Text>
+const SuperTextWithShadow = props => {
+  const fontSize = props.fontSize ? props.fontSize : 21
+  const fontFamily = props.fontFamily ? props.fontFamily : UPHEAVAL
```
Comment (EphemeralX) [inline civilapp/src/components/SuperText.js]: Thought we switched to Sportsball for everything?

File: civilapp/src/components/SuperText.js [hunk 8]
```diff
@@ -920,128 +477,45 @@ class RetroTextInput extends Component {
...
-            { width, fontSize, paddingTop: 10, right: -10 },
-          ]}
-        >
-          {value}
-        </Text>
+          <Text style={[styles.numberTextInputText, { fontSize, color: '#fff', zIndex: 1 }]}>{value}</Text>
+        )}
+        <Text style={[styles.numberTextInputText, { width, fontSize, right: -7, paddingTop: 7 }]}>{value}</Text>
+        <Text style={[styles.numberTextInputText, { width, fontSize, left: -7, paddingBottom: 7 }]}>{value}</Text>
+        <Text style={[styles.numberTextInputText, { width, fontSize, left: -7, paddingBottom: 7 }]}>{value}</Text>
```
Comment (EphemeralX) [inline civilapp/src/components/SuperText.js]: These 2 lines are identical, is that intentional?

**Comment by rauchp** [comment]
The first screenshot shows how the keyboard keys were overlapping with the footer on Android? I know that text is supposed to have drop shadows, that's why I added "(pre-fixing SuperText)" to highlight that the image only shows the fix of the layout of that page.

**Comment by rauchp** [comment]
Yeah, we can smooth it out with the shadowSize props in the new SuperText. I can do that, but wouldn't that fit better for a separate polishing PR, rather than right now also go through the 70+ times we use the super text component and make sure the smoothing and width is right?

File: civilapp/ios/Podfile.lock (hunk 1 above)
Comment (rauchp) [reply civilapp/ios/Podfile.lock]: It is? https://github.com/outlier-collective/mono/blob/master/civilapp/.gitignore#L75-L79
I'll have to see why it still pushed

**Comment by rauchp** [comment]
still pixel tweaking, redid drop shadows. 

Screen Shot 2019-12-27 at 9 22 19 PM


got the room view header text to look like figma too but working on simplifying code rn.

File: civilapp/src/components/SuperText.js (hunk 7 above)
Comment (rauchp) [reply civilapp/src/components/SuperText.js]: u right

File: civilapp/src/components/SuperText.js (hunk 8 above)
Comment (rauchp) [reply civilapp/src/components/SuperText.js]: getting nitted for code I haven't touched in months is next gen nitting

**Pull request merged**